python_scripts/cross_validation_learning_curve.py

```diff
@@ -102,10 +102,11 @@
 # benefit to adding samples anymore or assessing the potential gain of adding
 # more samples into the training set.
 #
-# If we achieve a plateau and adding new samples in the training set does not
-# reduce the testing error, we might have reached the Bayes error rate using the
-# available model. Using a more complex model might be the only possibility to
-# reduce the testing error further.
+# If the testing error plateaus despite adding more training samples, it's
+# possible that the model has achieved its optimal performance. In this case,
+# using a more expressive model might help reduce the error further. Otherwise,
+# the error may have reached the Bayes error rate, the theoretical minimum error
+# due to inherent uncertainty in the data.
```

Comment:
It might be helpful to explain why this is expected:

```suggestion
# due to inherent uncertainty not resolved by the available data. This minimum error is
# non-zero whenever some of the variation of the target variable `y` depends on
# external factors not fully observed in the features available in `X`, which is
# almost always the case in practice.
```